OMCompiler/Compiler/NBackEnd/Util/NBBackendUtil.mo

```diff
@@ -49,15 +49,28 @@ public
   // Util imports
   import Util;
 
+  function find
```

Comment:
I think this name is far too generic. Maybe `reduceBoolArrayToIndices` is too revealing but there must be something in between :sweat_smile:

Reply:
It could also be implemented simply as:
```Modelica
indices := list(i for i guard arr[i] in arrayLength(arr):-1:1);
```
Or `1:arrayLength(arr)` if the result list doesn't need to be reversed.

Reply:
I just used `find` because that is the name for this function in matlab, but i am fine with other names. Also i was not aware that you can actually guard a for loop inside a list constructor. Is that efficient? I would use it a ton!

Reply:
Yes, it's very efficient as long as the test is efficient. You can easily write filter+map operations using the list operator.

Reply:
Awesome!